Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabPublicHandler.cpp

```diff
@@ -566,6 +566,20 @@ namespace AzToolsFramework
 
                 CreateLink(instanceToCreate->get(), instanceToParentUnder->get().GetTemplateId(), undoBatch.GetUndoBatch(), AZStd::move(patch));
 
+                // Update the parent instance cache so that the new instance doesn't get recreated during propagation
+                if (PrefabDomReference cachedOwningInstanceDom = instanceToParentUnder->get().GetCachedInstanceDom();
+                    cachedOwningInstanceDom.has_value())
+                {
+                    const PrefabDom& targetTemplatePrefabDom = m_prefabSystemComponentInterface->FindTemplateDom(instanceToParentUnder->get().GetTemplateId());
+                    PrefabDomPath instancePath = PrefabDomUtils::GetPrefabDomInstancePath(instanceToCreate->get().GetInstanceAlias().c_str());
+                    const PrefabDomValue* instanceValue = instancePath.Get(targetTemplatePrefabDom);
+                    if (instanceValue)
+                    {
+                        PrefabDomUtils::AddOrUpdateNestedInstance(
+                            cachedOwningInstanceDom->get(), instanceToCreate->get().GetInstanceAlias(), *instanceValue);
```

Comment:
If you are instantiating a new instance, then there should be no need to 'update the nested instance'. Instead of doing AddOrUpdateNestedInstance, I recommend changing this to simply add the nested instance dom to the cached dom.

Reply:
Changed the function name to "AddNestedInstance", and assert if instance member already exists. However, with that I had to revert the changes in PrefabSystemComponent to use the new helper because the original logic there only wants to add the instance to the template DOM if it doesn't already exist. There are cases in CreatePrefab where the instance already exists in the template DOM (not the cached instance DOM).